sklearn/tree/_utils.pyx

```diff
@@ -240,6 +206,38 @@ cdef class PriorityHeap:
     cdef bint is_empty(self) nogil:
         return self.heap_ptr <= 0
 
+    cdef void heapify_up(self, PriorityHeapRecord* heap, SIZE_t pos) nogil:
```

Comment:
Yeah, I originally moved it in because I made another priorityheap class for use with non-records, but we ended up deciding that was too slow. I do think it is cleaner, though.